Sources/BraintreeCore/ConfigurationLoader.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+
+class ConfigurationLoader {
+    
+    // MARK: - Private Properties
+    
+    private let configPath = "v1/configuration"
+    private let configurationCache: ConfigurationCache = ConfigurationCache.shared
+    private let http: BTHTTP
+    
+    init(http: BTHTTP) {
+        self.http = http
+    }
+    
+    deinit {
+        http.session.finishTasksAndInvalidate()
+    }
+    
+    func getConfig(_ authorization: ClientAuthorization, completion: @escaping (BTConfiguration?, Error?) -> Void) {
+        // Fetches or returns the configuration and caches the response in the GET BTHTTP call if successful
```

Comment:
Nit - this should probably be a docstring instead of a comment

Reply:
Thats a good idea, updated: [418b39d](https://github.com/braintree/braintree_ios/pull/1353/commits/418b39d4f9c799335b6393f2fd9195fa034a674f)